apps/meteor/client/components/Omnichannel/CustomFieldsForm/CustomSelectInput.tsx

```diff
@@ -0,0 +1,48 @@
+import { Field, Select } from '@rocket.chat/fuselage';
+import type { OmnichannelCustomFieldEndpointPayload } from '@rocket.chat/rest-typings';
+import { useTranslation } from '@rocket.chat/ui-contexts';
+import React, { forwardRef, ReactElement, useMemo } from 'react';
+import { UseControllerReturn, useFormContext } from 'react-hook-form';
+
+export const CustomSelectInput = forwardRef<
+	HTMLInputElement,
+	UseControllerReturn['field'] & {
+		data: OmnichannelCustomFieldEndpointPayload;
+	}
+>(function CustomSelectInput({ data, name, value, onChange }, ref): ReactElement {
+	const t = useTranslation();
+	const {
+		formState: { errors },
+	} = useFormContext();
+
+	const mappedOptions = useMemo(() => data.options.split(',').map<[string, string]>((value) => [value, value]), [data.options]);
+
+	const errorMessage = useMemo(() => {
+		if (errors?.livechatData && errors?.livechatData[data._id]?.type === 'required') {
+			return t('The_field_is_required', data.label);
+		}
+	}, [data._id, data.label, errors?.livechatData, t]);
+
+	return (
+		<Field>
+			<Field.Label>
+				{data.label}
+				{data.required && '*'}
+			</Field.Label>
+			<Field.Row>
```

Comment:
Have you considered whether or not all this logic regarding field validation, label, etc could be its own component shared by `CustomSelectInput` and `CustomTextInput`?

Reply:
Yes, but... since we have different type of components, I think is better to do it separately...

Reply:
well... I did something 😬